lib/mongodb.js

```diff
@@ -744,26 +749,42 @@ function buildNearFilter(query, near) {
     coordinates = near.near;
   }
 
-  query.where[near.key] = {
-    near: {
-      $geometry: {
-        coordinates: [coordinates.lng, coordinates.lat],
-        type: 'Point',
-      },
+  node.near = {
+    $geometry: {
+      coordinates: [coordinates.lng, coordinates.lat],
+      type: 'Point',
     },
   };
-};
+
+  if (near.maxDistance) {
+    node.near['$maxDistance'] = near.maxDistance;
+  }
+}
 
 function hasNearFilter(where) {
   if (!where) return false;
 
-  for (var k in where) {
-    if (where[k] && typeof where[k] === 'object' && where[k].near) {
-      return true;
+  var found = false;
+  var keyPath = [];
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      var keys = Object.keys(node);
+      for (var i = 0; i < keys.length; i++) {
+        var ex = node[keys[i]];
+        keyPath.push(keys[i]);
+        if (ex && ex.near) {
```

Comment:
@superkhau sorry, "inherited" was probably not clear in this context... I meant that the code was written as such by the prior contributor and I simply augmented it with the intention of not changing anything that was relevant to the fix. I updated the variable name to `nodeProperty` - let me know if that's still not clear. Thanks.